pkg/controller/config/config_controller.go

```diff
@@ -292,42 +294,72 @@ func (r *ReconcileConfig) validatePipeline(req reconcile.Request, cfg *op.Config
 	log := requestLogger(req, "validate-pipeline")
 	log.Info("validating pipelines")
 
+	running, err := r.validateDeployments(req, cfg)
+	if err != nil {
+		log.Error(err, "failed to validate pipeline controller deployments")
+		_ = r.updateStatus(cfg, op.ConfigCondition{
+			Code:    op.PipelineValidateError,
+			Details: err.Error(),
+			Version: flag.TektonVersion})
+		return reconcile.Result{}, err
+	}
+
+	if !running {
+		return reconcile.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
+	}
+
+	found, err := validate.Webhook(context.TODO(), r.client, flag.PipelineWebhookConfiguration)
+	if err != nil {
+		log.Error(err, "failed to validate mutating webhook")
+		_ = r.updateStatus(cfg, op.ConfigCondition{
+			Code:    op.PipelineValidateError,
+			Details: err.Error(),
+			Version: flag.TektonVersion})
+		return reconcile.Result{RequeueAfter: 15 * time.Second}, err
+	}
+	if !found {
+		return reconcile.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
```

Comment:
What's the intent of requeing here again? :thinking:

Reply:
@hrishin https://github.com/openshift/tektoncd-pipeline-operator/pull/143#discussion_r357475469

Reply:
so if the deployment is not available yet, we need to check again, preferably a timed requeue than an a exponential requeue. So when tekton controller and tekton-webhook is available we can mark config condition `validated` and proceed with installing addons.

Reply:
Thanks  @nikhil-thomas for the explaining it :+1: . However wondering that change in state of the deployment would trigger reconciliation, thus do we need the requeue explicitely? 

Though nothing harmful as such on queueing it again, rather its guarantee that reconciliation loop would trigger again.